### PR TITLE
Add missing project fields

### PR DESCRIPTION
**Pull request by rgolangh**
Signed-off-by: Roy Golan <rgolan@redhat.com>

### CODE REVIEW

**Review by RichardW98 — APPROVED** [review]

File: plugins/parodos/src/models/project.ts [hunk 1]
```diff
@@ -8,17 +8,26 @@ const projectStatus = z.union([
...
-  createDate: z.coerce.date(),
-  modifyDate: z.coerce.date(),
+  createdDate: z.coerce.date(),
+  modifiedDate: z.coerce.date(),
   status: projectStatus
     .default('IN_PROGRESS')
     .transform(value => value.split('_').join(' ')),
+  modifiedBy: z.string().nullable(),
+  createdBy: z.string().nullable(),
+  accessRole: accessRole.transform(value => value.split('_').join(' ')),
```
Comment (dagda1) [inline plugins/parodos/src/models/project.ts:30]:
```suggestion
  accessRole,
```
I don't think we need the transform as none of the elements of the `accessRole` string union  have a `_` in them.

File: plugins/parodos/src/models/project.ts [hunk 2]
```diff
@@ -8,17 +8,26 @@ const projectStatus = z.union([
...
   name: z.string(),
   description: z.string().nullable(),
-  createDate: z.coerce.date(),
-  modifyDate: z.coerce.date(),
+  createdDate: z.coerce.date(),
+  modifiedDate: z.coerce.date(),
   status: projectStatus
     .default('IN_PROGRESS')
     .transform(value => value.split('_').join(' ')),
+  modifiedBy: z.string().nullable(),
```
Comment (dagda1) [inline plugins/parodos/src/models/project.ts:28]:
```suggestion
  modifiedBy: z.string().nullable().optional(),
```

Everything is required by default unless  `optional` is specified.

File: plugins/parodos/src/models/project.ts [hunk 3]
```diff
@@ -8,17 +8,26 @@ const projectStatus = z.union([
   z.literal('COMPLETED'),
 ]);
 
+const accessRole = z.union([
```
Comment (dagda1) [inline plugins/parodos/src/models/project.ts:11]: good work!

**Review by dagda1 — CHANGES_REQUESTED** [review]

File: plugins/parodos/src/models/project.ts [hunk 4]
```diff
@@ -8,17 +8,26 @@ const projectStatus = z.union([
...
+  z.literal('Developer'),
+  z.literal('Admin'),
+]);
+
 export type ProjectStatus = z.infer<typeof projectStatus>;
 
 export const projectSchema = z.object({
   id: z.string(),
   name: z.string(),
   description: z.string().nullable(),
```
Comment (dagda1) [inline plugins/parodos/src/models/project.ts:22]:
```suggestion
  description: z.string().nullable().optional(),
```
Needs to be optional now as the API is no longer returning `null` for description.

**Comment by dagda1** [comment]
To make the tests pass, you need to change lines 50-51 in `plugins/parodos/src/components/TestApp.tsx` to

```ts
new Response(
  JSON.stringify([
    {
      id: '511da8ce-4df7-438b-a9ec-0130f14884bd',
      name: 'myProject',
      description: 'My description',
      createdDate: '2023-04-14T07:55:38.144+00:00',
      modifiedDate: '2023-04-14T07:55:38.144+00:00',
      createdBy: 'test',
      accessRole: 'Owner'
    },
  ]),
```

File: plugins/parodos/src/models/project.ts (hunk 1 above)
Comment (rgolangh) [reply plugins/parodos/src/models/project.ts:30]: Ack. I just thought to e future proof and standard with other enums.

File: plugins/parodos/src/models/project.ts (hunk 2 above)
Comment (rgolangh) [reply plugins/parodos/src/models/project.ts:28]: ack

File: plugins/parodos/src/models/project.ts (hunk 1 above)
Comment (dagda1) [reply plugins/parodos/src/models/project.ts]: that is a valid point.  Feel free to leave it in if you want

**Pull request merged**